### DISCUSSION TITLE
Why aren't most methods in FilesClient async?

### DISCUSSION THREAD

**Discussion by ingshak** [ubisoft/NGitLab#741] [Q&A] [answered]
Why aren't most methods in FilesClient async?

**Comment by louis-z** [comment, marked as answer]
Since NGitLab was created 10 years ago, i.e. before the async paradigm became a thing, its initial API was synchronous. In the last few years, however, we've gradually enhanced the API with async methods, but it's far from being complete.

We appreciate external contributions, so if you're up to it we encourage you to submit PRs to increase the API's async coverage.
